app/models/sync.rb

```diff
@@ -0,0 +1,68 @@
+require 'csv'
+
+class Sync
+  def self.apply
+    new.apply
+  end
+
+  def initialize
+    @file = file
+    @table = 'bearden.ranked'
+  end
+
+  def apply
+    export_csv
+
+    connection = Redshift.connect
+    connection.transaction do |conn|
+      truncate(conn)
+      copy(conn)
+    end
+
+  rescue PG::Error => e
+    return e.message
+
+  ensure
+    connection&.close
+  end
+
+  private
+
+  def truncate(conn)
+    conn.exec("TRUNCATE #{@table}")
+  end
+
+  def copy(conn)
+    source = "s3://#{ENV['AWS_BUCKET']}/#{@file}"
+    conn.exec("COPY #{@table} \
+                FROM '#{source}' \
+                WITH CREDENTIALS '#{Redshift.s3_auth}' \
+                DELIMITER ',' \
+                REGION '#{ENV['AWS_REGION']}' \
+                CSV IGNOREHEADER 1 EMPTYASNULL")
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def export_csv
```

Comment:
FYI, I'm working on extracting this functionality into an object we can reuse. I'll link it up when it done so that you can use it here. 🎩